delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/operations/s3/client/S3StorageClient.scala

```diff
@@ -1,18 +1,31 @@
 package ch.epfl.bluebrain.nexus.delta.plugins.storage.storages.operations.s3.client
 
+import akka.http.scaladsl.model.Uri
 import cats.effect.{IO, Resource}
+import cats.implicits.toBifunctorOps
 import ch.epfl.bluebrain.nexus.delta.plugins.storage.storages.StoragesConfig.S3StorageConfig
 import ch.epfl.bluebrain.nexus.delta.sdk.error.ServiceError.FeatureDisabled
+import eu.timepit.refined.collection.NonEmpty
+import eu.timepit.refined.refineV
+import fs2.aws.s3.S3
+import fs2.aws.s3.models.Models.{BucketName, FileKey}
 import io.laserdisc.pure.s3.tagless.{Interpreter, S3AsyncClientOp}
 import software.amazon.awssdk.auth.credentials.{AwsBasicCredentials, StaticCredentialsProvider}
 import software.amazon.awssdk.regions.Region
 import software.amazon.awssdk.services.s3.S3AsyncClient
 import software.amazon.awssdk.services.s3.model.{ListObjectsV2Request, ListObjectsV2Response}
+import fs2.Stream
 
 import java.net.URI
 
 trait S3StorageClient {
   def listObjectsV2(bucket: String): IO[ListObjectsV2Response]
+
+  def readFile(bucket: String, fileKey: String): fs2.Stream[IO, Byte]
+
+  def underlyingClient: S3AsyncClientOp[IO]
+
+  def baseEndpoint: IO[Uri]
```

Comment:
Purely for convenience, ideally we shouldn't need to expose these here, it's a smell. All S3 interactions can be abstracted by the client once we know exactly the calls we need w/ the correct parameters.